iroh/src/commands/delete.rs

```diff
@@ -0,0 +1,29 @@
+use anyhow::Result;
+use clap::Subcommand;
+use iroh::{client::quic::RpcClient, rpc_protocol::SetRootRequest};
+
+#[derive(Subcommand, Debug, Clone)]
+pub enum Commands {
+    /// Delete the given roots
+    Roots { name: String },
+}
+
+impl Commands {
+    pub async fn run(self, client: RpcClient) -> Result<()> {
+        match self {
+            Commands::Roots { name } => {
+                let name = if name.starts_with("\"") && name.ends_with("\"") && name.len() > 1 {
+                    name[1..name.len() - 1].as_bytes().to_vec()
+                } else {
+                    hex::decode(name)?
+                }
+                .into();
+                let response = client.rpc(SetRootRequest { name, value: None }).await?;
```

Comment:
I'm assuming that when you set the value to `None`, that is how you mark it for deletion?

It's a bit confusing because I'm now associating the word "root" w/ what we keep, but it looks like it can also be associated with what we delete.

I think a better way of describing my confusion is, it's hard to tell what you are actually doing by sending this `SetRootRequest` unless you already have an understanding of the whole system. Might be solved with a comment, but might be work a re-name

Reply:
Yes, I did not want to have a full set of requests for a CRUD API, so I did this, a request that can both set and delete.

Reply:
We are going to rename roots to tags.